src/Roassal/RSMorphicHost.class.st

```diff
@@ -83,11 +83,13 @@ RSMorphicHost >> dirtyRectangle [
 
 { #category : 'accessing' }
 RSMorphicHost >> dirtyRectangle: aRectangleOrNil [
+
 	| visibleRectangle |
 	dirtyRectangle := aRectangleOrNil.
 	dirtyRectangle ifNil: [ ^ self ].
 	visibleRectangle := canvas visibleRectangle.
-	dirtyRectangle := dirtyRectangle intersect: visibleRectangle
+	"We already do the same check above but this second guard is here to workaround a concurrency problem..."
+	dirtyRectangle := dirtyRectangle ifNotNil: [ :dr | dr intersect: visibleRectangle ]
```

Comment:
Aha, so the `dirtyRectangle` changes while the `canvas visibleRectangle` is calculated.

Reply:
This mostly happens in our CI with the tests of a Moose browser including a visualization. I think Clotilde found a way to reproduce it locally, but it's still random. Like she found a way to reproduce it while spamming the "run tests" button